plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt_test.go

```diff
@@ -96,6 +96,166 @@ func TestHandleTelemetryTwoSimple(t *testing.T) {
+										},
+										{
+											Name:        prefixLength.name,
+											ValueByType: &telemetryBis.TelemetryField_Uint32Value{Uint32Value: prefixLength.uint32Value},
+										},
+										{
+											Name:        origin.name,
+											ValueByType: &telemetryBis.TelemetryField_StringValue{StringValue: origin.stringValue},
+										},
+										{
+											Name:        status.name,
+											ValueByType: &telemetryBis.TelemetryField_StringValue{StringValue: status.stringValue},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		tags: map[string]string{
+			encodingPath.name: encodingPath.stringValue,
+			name.name:         name.stringValue,
+			index.name:        index.stringValue,
+			ip.name:           ip.stringValue,
+			source.name:       source.stringValue,
+			subscription.name: subscription.stringValue,
+		},
+		fields: map[string]interface{}{
+			deleteKey:              false,
+			ip.fieldName:           ip.stringValue,
+			prefixLength.fieldName: prefixLength.uint64Value,
+			origin.fieldName:       origin.stringValue,
+			status.fieldName:       status.stringValue,
+		},
+	},
+		{
+			telemetry: &telemetryBis.Telemetry{
+				MsgTimestamp: 1543236572000,
+				EncodingPath: encodingPath.stringValue,
+				NodeId:       &telemetryBis.Telemetry_NodeIdStr{NodeIdStr: source.stringValue},
+				Subscription: &telemetryBis.Telemetry_SubscriptionIdStr{SubscriptionIdStr: subscription.stringValue},
+				DataGpbkv: []*telemetryBis.TelemetryField{
+					{
+						Delete: true,
+						Fields: []*telemetryBis.TelemetryField{
+							{
+								Name: "keys",
+								Fields: []*telemetryBis.TelemetryField{
+									{
+										Name:        name.name,
+										ValueByType: &telemetryBis.TelemetryField_StringValue{StringValue: name.stringValue},
+									},
+									{
+										Name:        index.name,
+										ValueByType: &telemetryBis.TelemetryField_Uint32Value{Uint32Value: index.uint32Value},
+									},
+									{
+										Name:        ip.name,
+										ValueByType: &telemetryBis.TelemetryField_StringValue{StringValue: ip.stringValue},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			tags: map[string]string{
+				encodingPath.name: encodingPath.stringValue,
+				name.name:         name.stringValue,
+				index.name:        index.stringValue,
+				ip.name:           ip.stringValue,
+				source.name:       source.stringValue,
+				subscription.name: subscription.stringValue,
+			},
+			fields: map[string]interface{}{deleteKey: true},
+		},
+	}
+	for _, test := range testCases {
+		c := &CiscoTelemetryMDT{
+			Log:                testutil.Logger{},
+			Transport:          "dummy",
+			Aliases:            map[string]string{"deleted": encodingPath.stringValue},
+			IncludeDeleteField: true}
+		acc := &testutil.Accumulator{}
+		err := c.Start(acc)
+		// error is expected since we are passing in dummy transport
+		require.Error(t, err)
```

Comment:
Can we please reflect this in the `require` statement
```suggestion
		// error is expected since we are passing in dummy transport
		require.ErrorContains(t, c.Start(acc), "a message part you expect")
```